### PR TITLE
SE-846-Modify-Implementation-Strategy-to-Support-only-IProtocolToken

### PR DESCRIPTION
**Pull request by Palmerg4**

### CODE REVIEW

File: docs/OVERVIEW.md [hunk 1]
```diff
@@ -56,8 +56,8 @@ This document offers a high level overview of the rules protocol - what it does
...
-| Protocol Supported ERC721 | An ERC721 token that implements the protocol ProtocolERC721 or the protocol hook.                                                                                                |
-| Protocol Supported ERC20  | An ERC20 token that implements the protocol ProtocolERC20 or the protocol hook.                                                                                                |
+| Protocol Supported ERC721 | An ERC721 token that implements the protocol ApplicationERC721 or the protocol hook.                                                                                                |
+| Protocol Supported ERC20  | An ERC20 token that implements the protocol ApplicationERC20 or the protocol hook.                                                                                                |
```
Comment (ShaneDuncan602) [inline docs/OVERVIEW.md]: I think that we should say, "An ERC721 token that implements the IProtocolToken.sol and contains the protocol hook. "

File: .gitmodules [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [submodule "lib/openzeppelin-contracts"]
 	path = lib/openzeppelin-contracts
-	url = https://github.com/openzeppelin/openzeppelin-contracts
+	url = https://github.com/OpenZeppelin/openzeppelin-contracts
```
Comment (ShaneDuncan602) [inline .gitmodules]: Not really sure why this would change.

File: docs/userGuides/architecture/client/PROTOCOL-ERC-20.md
Comment (ShaneDuncan602) [inline docs/userGuides/architecture/client/PROTOCOL-ERC-20.md:1]: I believe this file name should change.

File: docs/userGuides/development/invariants/client/token/ERC20/ProtocolERC20-invariants.md
Comment (ShaneDuncan602) [inline docs/userGuides/development/invariants/client/token/ERC20/ProtocolERC20-invariants.md:1]: this file name should change as well

File: docs/userGuides/rules/CUSTOMIZATIONS.md [hunk 3]
```diff
@@ -42,9 +42,9 @@ To activate a deactivated rule, invoke the activation function and send it a par
          ````
 
 ### NFT Functionality
-Protocol supported NFT's implement OpenZeppelin ERC721Burnable, ERC721URIStorage, and ERC721Enumerable. These interfaces may be utilized in application NFT's. 
+Protocol supported NFT's implement OpenZeppelin AccessControl, ERC721, ReetrancyGuard, and AccessControl. These interfaces may be utilized in application NFT's.
```
Comment (ShaneDuncan602) [inline docs/userGuides/rules/CUSTOMIZATIONS.md]: AccessControl is listed twice. The second should be ERC721Enumerable

File: src/example/ERC721/ApplicationERC721.sol [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+import "src/client/token/IProtocolTokenHandler.sol";
+import {IZeroAddressError} from "src/common/IErrors.sol";
+
+/**
+ * @title Example ERC721 ApplicationERC721
+ * @author @ShaneDuncan602, @oscarsernarosero, @TJ-Everett, @Palmerg4
+ * @notice This is an example implementation that App Devs should use.
+ * @dev During deployment _tokenName _tokenSymbol _tokenAdmin are set in constructor
+ */
+contract ApplicationERC721 is ERC721, AccessControl, IProtocolToken, IZeroAddressError, ReentrancyGuard, ERC721Burnable {
```
Comment (ShaneDuncan602) [inline src/example/ERC721/ApplicationERC721.sol]: This needs to have enumerable as well. Some rules won't work without it. I'm a little concerned those tests haven't been failing without it.

File: test/util/UtilProtocolERC721.sol [hunk 5]
```diff
@@ -14,10 +14,10 @@ import "src/client/token/handler/diamond/IHandlerDiamond.sol";
 /**
  * @title Protocol ERC721 Base Contract
  * @author @ShaneDuncan602, @oscarsernarosero, @TJ-Everett
- * @notice This is the base contract for all protocol ERC721s
+ * @notice This is a base contract for all test ERC721s
  */
 
-contract ProtocolERC721 is ERC721Burnable, ERC721URIStorage, ERC721Enumerable, ProtocolTokenCommon, AppAdministratorOrOwnerOnly, ReentrancyGuard {
+contract UtilProtocolERC721 is ERC721Burnable, ERC721URIStorage, ERC721Enumerable, ProtocolTokenCommon, AppAdministratorOrOwnerOnly, ReentrancyGuard {
```
Comment (ShaneDuncan602) [inline test/util/UtilProtocolERC721.sol:20]: Ah, this is why the tests are failing. This includes enumerable. We probably want to add it to the ApplicationERC721 as well though.

**Review by ShaneDuncan602 — CHANGES_REQUESTED** [review]
Great job here. Just found a few things. Mostly around docs and Enumerable.

File: docs/userGuides/development/invariants/client/token/ERC721/ProtocolERC721-invariants.md
Comment (TJ-Everett) [inline docs/userGuides/development/invariants/client/token/ERC721/ProtocolERC721-invariants.md:1]: This file name should change too I believe

**Review by TJ-Everett — COMMENTED** [review]
Awesome work on this Gordon. Really really close just the one file name for me and then Shane's feedback.

**Review by TJ-Everett — APPROVED** [review]
This looks good to go to me!

**Pull request merged**